src/xdisplay.cpp

```diff
@@ -125,11 +82,110 @@ namespace xpyt
         return pub_data;
     }
 
+    /*******************************
+     * xdisplayhook implementation *
+     *******************************/
+
+    xdisplayhook::xdisplayhook()
+        : m_execution_count(0)
+    {
+    }
+
+    xdisplayhook::~xdisplayhook()
+    {
+    }
+
+    void xdisplayhook::set_execution_count(int execution_count)
+    {
+        m_execution_count = execution_count;
+    }
+
+    void xdisplayhook::operator()(const py::object& obj, bool raw = false) const
+    {
+        auto& interp = xeus::get_interpreter();
+
+        if (!obj.is_none())
+        {
+            if (hasattr(obj, "_ipython_display_"))
+            {
+                obj.attr("_ipython_display_")();
+                return;
+            }
+
+            nl::json pub_data;
+            if (raw)
+            {
+                pub_data = obj;
+            }
+            else
+            {
+                pub_data = mime_bundle_repr(obj);
+            }
+
+            interp.publish_execution_result(
+                m_execution_count,
+                std::move(pub_data),
+                nl::json::object()
+            );
+        }
+    }
+
+    /***************************
+     * xdisplay implementation *
+     ***************************/
+
+    void xdisplay(const py::object& obj, const py::object display_id, bool update)
```

Comment:
The display function differs from the display hook in that it does not use `publish_execution_result` but `display_data` and `update_display_data`, and therefore does not produce an output prompt.